packages/tracing/tests/helpers/tracesUtils.ts

```diff
@@ -14,11 +14,16 @@ import {
   expectedCustomResponseValue, 
   expectedCustomErrorMessage,
 } from '../e2e/constants';
+import { FunctionSegmentNotDefinedError } from './FunctionSegmentNotDefinedError';
 interface ParsedDocument {
   name: string
   id: string
   start_time: number
   end_time: number
+  // This flag may be set if the segment hasn't been fully processed
+  // The trace may have already appeared in the `getTraceSummaries` response 
+  // but a segment may still be in_progress
+  in_progress?: boolean 
```

Comment:
By looking at the documentation, also the `end_time` key might be not defined, correct? 

> end_time – number that is the time the segment was closed. For example, 1480615200.090 or 1.480615200090E9. Specify either an end_time or in_progress.

https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-fields

Reply:
Yes. In my experiments, `end_time` field doesn't exist.  It should be optional.

<img width="506" alt="image" src="https://user-images.githubusercontent.com/61999/167675572-371d4ecf-c73e-4695-bd15-02146f11e219.png">